app/lib/installer.js

```diff
@@ -0,0 +1,336 @@
+/**
+ * @module dataserv-client/installer
+ */
+
+'use strict';
+
+var EventEmitter = require('events').EventEmitter;
+var inherits = require('util').inherits;
+var os = require('os');
+var Logger = require('./logger');
+var exec = require('child_process').exec;
+var remote = require('remote');
+var app = remote.require('app');
+var request = require('request');
+var fs = require('fs-extra');
+var ZipFile = require('adm-zip');
+var path = require('path');
+
+/**
+ * Represents a dataserv-client installer
+ * @constructor
+ */
+function DataServInstaller() {
+  if (!(this instanceof DataServInstaller)) {
+    return new DataServInstaller();
+  }
+
+  this._logger = new Logger();
+  this._platform = os.platform();
+  this._userdir = app.getPath('userData');
+  this._destination = this._userdir + '/tmp/dataserv-client.zip';
+
+  this._targets = {
+    linux: {
+      install: this._installGnuLinux.bind(this),
+      check: this._checkGnuLinux.bind(this),
+      path: 'dataserv-client'
+    },
+    darwin: {
+      install: this._installMacintosh.bind(this),
+      check: this._checkMacintosh.bind(this),
+      path: this._userdir +
+            '/dataserv-client.app/Contents/MacOS/dataserv-client'
+    },
+    win32: {
+      install: this._installWindows.bind(this),
+      check: this._checkWindows.bind(this),
+      path: this._userdir + '\\dataserv-client\\dataserv-client.exe'
+    }
+  };
+}
+
+inherits(DataServInstaller, EventEmitter);
+
+/**
+ * Initializes the installer and begins emitting events
+ * #install
+ * @param {String} password - gnu/linux only
+ */
+DataServInstaller.prototype.install = function(password) {
+  if (Object.keys(this._targets).indexOf(this._platform) === -1) {
+    return this.emit('error', new Error('This platform is not supported'));
+  }
+
+  var self = this;
```

Comment:
:+1: 

Reviewing this code almost brought a tear to my eyes. This is a much more coherent way of structuring the project, lots of improvements, very happy!